### PR TITLE
Pregenerate rounding code and call it

### PR DESCRIPTION
**Pull request by hrydgard**
Greatly reduces code bloat when parts of the JIT are off, or in general for stuff that falls back to the interpreter (like currently LWL/LWR on ARM64). Possibly a minor speed improvement (due to reduced instruction cache pressure) but mostly reduces the chance of filling up the jit cache, and makes the disassembly easier to read for debugging purposes.

This is a redoing of #8036 , but with the cleaned-up code.

### CODE REVIEW

File: Core/MIPS/ARM/ArmAsm.cpp [hunk 1]
```diff
@@ -70,9 +71,84 @@ namespace MIPSComp {
...
 using namespace ArmJitConstants;
 
-void ArmJit::GenerateFixedCode()
-{
-	enterCode = AlignCode16();
+void ArmJit::GenerateFixedCode() {
+	const u8 *start = GetCodePtr();
+
+	// LR == SCRATCHREG2....  Maybe we could find some other register to use to avoid having to
+	// push LR?
```
Comment (unknownbrackets) [inline Core/MIPS/ARM/ArmAsm.cpp]: I wonder... maybe we just enforce that other regs must be flushed?  As long as it's always after a FlushAll (might already be), we could use other regs normally used for alloc.

-[Unknown]

Comment (unknownbrackets) [inline Core/MIPS/ARM/ArmAsm.cpp]: Then again, isn't POP R_PC the preferred way to return?  Maybe it's good, and the PUSH of SCRATCHREG1 should combine, right?

-[Unknown]

File: Core/MIPS/x86/Jit.cpp [hunk 2]
```diff
@@ -508,7 +455,7 @@ bool Jit::DescribeCodePtr(const u8 *ptr, std::string &name)
 			snprintf(temp, sizeof(temp), "%08x", jitAddr);
 		name = temp;
 	}
-	else if (asm_.IsInSpace(ptr))
+	else if (IsInSpace(ptr))
```
Comment (unknownbrackets) [inline Core/MIPS/x86/Jit.cpp]: Well, this is a duplicate to `name = "Unknown";` below.  Hopefully that never happens, but we only need one of the two.

-[Unknown]

**Comment by unknownbrackets** [comment]
Should we wait for 1.1.1 to merge this, or merge it now?  Just worried if there's any additional unexpected breakage...

-[Unknown]

File: Core/MIPS/ARM/ArmAsm.cpp (hunk 1 above)
Comment (hrydgard) [inline Core/MIPS/ARM/ArmAsm.cpp]: Yeah, it's no big deal really. I'll probably just remove this comment.

We do only call it when flushed, I think, so you're right that we could probably just grab R2 or whatever, but I probably won't bother for now.

File: Core/MIPS/x86/Jit.cpp (hunk 2 above)
Comment (hrydgard) [inline Core/MIPS/x86/Jit.cpp]: oh, oops. I'll fix it up.

**Comment by hrydgard** [comment]
@unknownbrackets  This commit by itself is not very dangerous, it just really moves where the rounding changes happen. I would be more worried about the other stuff I already merged. On the other hand, that stuff simply changed so that we do the same thing on ARM as on x86 (apart from removing the options), so should be fine (of course, disregarding potential bugs..).

I'll test as much as I can before pushing out 1.1.1.

**Comment by unknownbrackets** [comment]
Okay, I'll test some too later on various platforms, you're right that this should be fairly safe.

-[Unknown]

**Pull request merged**